### PR TITLE
chore: Add coordinated omission mode

### PR DESCRIPTION
**Pull request by romange**
With `--qps=0` dfly_bench will wait for the last request to be replied before sending the next one.

In addition, we add SEQUENTIAL mode that allows running requests sequentially through the keyspace.
It's not useful yet since each driver will send exactly the same sequence, but I will fix it in subsequent PRs.

### CODE REVIEW

File: src/server/dfly_bench.cc [hunk 1]
```diff
@@ -272,15 +273,21 @@ void Driver::Run(uint32_t num_reqs, uint64_t cycle_ns, ClientStats* dest) {
   for (unsigned i = 0; i < num_reqs; ++i) {
     int64_t now = absl::GetCurrentTimeNanos();
 
-    int64_t sleep_ns = next_invocation - now;
-    if (sleep_ns > 0) {
-      VLOG(5) << "Sleeping for " << sleep_ns << "ns";
-      ThisFiber::SleepFor(chrono::nanoseconds(sleep_ns));
+    if (cycle_ns) {
+      int64_t sleep_ns = next_invocation - now;
```
Comment (romange) [inline src/server/dfly_bench.cc:277]: jsu moved into the "if block"

File: src/server/dfly_bench.cc [hunk 2]
```diff
@@ -272,15 +273,21 @@ void Driver::Run(uint32_t num_reqs, uint64_t cycle_ns, ClientStats* dest) {
...
+        VLOG(5) << "Behind QPS schedule";
+      }
+      next_invocation += cycle_ns;
     } else {
-      VLOG(5) << "Behind QPS schedule";
-    }
-    next_invocation += cycle_ns;
+      // Coordinated omission.
 
+      fb2::NoOpLock lk;
```
Comment (romange) [inline src/server/dfly_bench.cc:288]: when cycle_ns is 0 we are in coordinated omission mode.

**Comment by romange** [comment]
@chakaz  PTAL

File: src/server/dfly_bench.cc [hunk 3]
```diff
@@ -207,34 +214,32 @@ class TLocalClient {
...
   CHECK_GT(range_, 0u);
 
-  if (dist == "U") {
-    dist_type_ = UNIFORM;
-  } else if (dist == "N") {
-    dist_type_ = NORMAL;
-    uint64_t stddev = GetFlag(FLAGS_key_stddev);
-    if (stddev != 0) {
-      stddev_ = double(stddev) / double(range_);
+  switch (dist_type) {
```
Comment (romange) [inline src/server/dfly_bench.cc:225]: i moved dist_type parsing to main

**Review by chakaz — APPROVED** [review]

**Pull request merged**